CraftingRevisions/Patches/GameManager_Awake.cs

```diff
@@ -4,15 +4,41 @@
 
 namespace CraftingRevisions.Patches
 {
-	[HarmonyPatch(typeof(GameManager), "Awake")]
-	internal class GameManager_Awake
+	// load the blueprints when the crafting panel is initialized
+	// (needed to be this late otherwise we have no instance of BlueprintManager)
+	[HarmonyPatch(typeof(Panel_Crafting), nameof(Panel_Crafting.Initialize))]
+	internal class Panel_Crafting_Initialize
 	{
 		private static void Postfix()
 		{
-			try { BlueprintMapper.MapBlueprints(); }
-			catch (System.Exception e)
+			// call TLD LoadAllUserBlueprints
+			InterfaceManager.m_Instance.m_BlueprintManager.LoadAllUserBlueprints();
+		}
+	}
+
+
+	// patched into postfix BlueprintManager.LoadAllUserBlueprints
+	// (BlueprintManager.RemoveUserBlueprints is not guaranteed to get called)
+	[HarmonyPatch(typeof(Il2CppTLD.Gear.BlueprintManager), nameof(Il2CppTLD.Gear.BlueprintManager.LoadAllUserBlueprints))]
+	internal class BlueprintManager_LoadAllUserBlueprints_Postfix
+	{
+		private static void Postfix(Il2CppTLD.Gear.BlueprintManager __instance)
+		{
+			// check we have a HasSet and it has contents
+			if (BlueprintManager.jsonUserBlueprints != null && BlueprintManager.jsonUserBlueprints.Count > 0)
 			{
-				MelonLogger.Error($"Blueprint Mapping failed: {e}");
+				// loop over the items
+				foreach (string jsonUserBlueprint in BlueprintManager.jsonUserBlueprints)
+				{
+					// load the blueprint into the game
+					bool loaded = __instance.LoadUserBlueprint(jsonUserBlueprint);
+
+#warning TODO - there appears to be no way to define an icon for this blueprint (UserBlueprintData still a WIP?)
```

Comment:
Icons are loaded with a naming convention based on the gear result name. It was the same before the DLC.

Reply:
icons are working, but appears to only show for certain gearitems (e.g. bandage is fine but Prybar is not)